vertx-web/src/main/java/io/vertx/ext/web/sstore/impl/SessionImpl.java

```diff
@@ -125,11 +153,30 @@ public boolean isDestroyed() {
   }
 
   @Override
+  public boolean isRegenerated() {
+    return renewed;
+  }
+
+  @Override
+  public String oldId() {
+    return oldId;
+  }
+
+  public int version() {
+    return version;
+  }
+
+  public int incrementVersion() {
+    return ++version;
```

Comment:
It is called only from the event loop when saving the session, so no other object will interact with it.